components/match2/commons/matches_table.lua

```diff
@@ -0,0 +1,340 @@
+	end
+	if _args.matchsection then
+		--to-Do: test this
+		lpdbConditions = lpdbConditions .. ' AND [[match2bracketdata_sectionheader::' .. _args.matchsection .. ']]'
+	end
+
+	return lpdbConditions
+end
+
+function MatchesTable._header()
+	local header = mw.html.create('tr')
+		:addClass('HeaderRow')
+		:node(mw.html.create('th')
+			:addClass('divCell')
+			:attr('data-sort-type','isoDate')
+			:wikitext('Date')
+		)
+
+	if _args.hideround ~= 'true' then
+		local cell = mw.html.create('th')
+			:addClass('divCell')
+			:wikitext('Round')
+		if not _args.sortround then
+			cell:addClass('unsortable')
+		end
+		header:node(cell)
+	end
+
+	header
+		:node(mw.html.create('th')
+			:addClass('divCell')
+			:wikitext('Opponent')
+		)
+		:node(mw.html.create('th')
+			:addClass('divCell')
+			:css('width','50')
+			:wikitext('Score')
+		)
+		:node(mw.html.create('th')
+			:addClass('divCell')
+			:wikitext('vs. Opponent')
+		)
+
+	return header
+end
+
+function MatchesTable._row(match)
+	local matchHeader = match.match2bracketdata.header
+	if String.isEmpty(matchHeader) then
+		if _currentID == match.match2bracketid then
+			matchHeader = _matchHeader or match.match2bracketdata.sectionheader
+		else
+			matchHeader = match.match2bracketdata.sectionheader or _matchHeader
+			if String.isEmpty(matchHeader) then
+				matchHeader = _matchHeader
+			end
+		end
+	end
+	if String.isNotEmpty(matchHeader) then
+		_matchHeader = matchHeader
+	else
+		matchHeader = '&nbsp;'
+	end
+	_currentID = match.match2bracketid
+
+	--if the header is a default bracket header we need to convert it to proper display text
+	matchHeader = DisplayHelper.expandHeader(matchHeader)[1]
+
+	local row = mw.html.create('tr')
+		:addClass('Match')
+
+	local dateCell = mw.html.create('td')
+		:addClass('Date')
+
+	local dateDisplay
+	if Logic.readBool(match.dateexact) then
+		local countdownArgs = {}
+		if (not Logic.readBool(match.finished)) and Logic.readBool(_args.countdown) then
+			countdownArgs = match.stream or {}
+			countdownArgs.rawcountdown = 'true'
+		else
+			countdownArgs.rawdatetime = 'true'
+		end
+		countdownArgs.date = match.date .. _UTC
+		dateDisplay = Countdown._create(countdownArgs)
+	elseif _args.dateexact == 'true' then
+		dateCell
+			:css('text-align', 'center')
+			:css('font-style', 'italic')
+		dateDisplay = 'To be announced'
+	else
+		dateDisplay = mw.language.new('en'):formatDate('F j, Y', match.date)
+	end
+
+	dateCell
+		:node(HiddenSort.run(match.date))
+		:node(dateDisplay)
+
+	row:node(dateCell)
+
+	if not Logic.readBool(_args.hideround) then
+		local roundCell = mw.html.create('td')
+			:addClass('Round')
+		if String.isNotEmpty(matchHeader) then
+			if Logic.readBool(_args.shortedroundnames) then
+				for long, short in pairs(_ABBREVIATIONS) do
+					matchHeader = matchHeader:gsub(long, short)
+				end
+			end
+			roundCell:wikitext(matchHeader)
+		end
+		row:node(roundCell)
+	end
+
+	-- workaround for a lpdb bug
+	-- remove when it is fixed
```

Comment:
Describe the bug in the comment. Right now only you can decide when this can be removed, because no one else will know what this is about.